### PR TITLE
implement sched_getaffinity

### LINKED ISSUE #56: Implement `sched_getaffinity`
[`sched_getaffinity`](https://docs.rs/rsix/0.25.0/rsix/process/fn.sched_getaffinity.html) is now implemented in rsix, so mustang just needs to add code to use it in c-scape.

### PR DESCRIPTION
**Pull request by nivkner**
fixes #56, along with everything else.

### CODE REVIEW

**Comment by nivkner** [comment]
needs https://github.com/bytecodealliance/rustix/pull/109

**Comment by sunfishcode** [comment]
bytecodealliance/rustix#109 is now in rustix 0.26.2.

File: c-scape/src/lib.rs [hunk 1]
```diff
@@ -2126,10 +2126,30 @@ unsafe extern "C" fn sched_yield() -> c_int {
     0
 }
 
+pub fn set_cpu(cpu: usize, cpuset: &mut data::CpuSet) {
```
Comment (sunfishcode) [inline c-scape/src/lib.rs:2129]: Can this function be non-`pub`?

**Comment by sunfishcode** [comment]
Great! If you want, you could also update the Known Limitations section in the top-level README.md now that child processes are working :-).

**Pull request merged**